### PR TITLE
BugFix : cache problem when joining asso

### PR DESCRIPTION
**Pull request by Mai-LinhC**
Currently there's a problem when joining an association because we're trying to get the information from the cache except it's not yet in the cache since we're just joining...

### CODE REVIEW

File: app/src/main/java/com/github/se/assocify/ui/screens/createAssociation/CreateAssociationViewmodel.kt [hunk 1]
```diff
@@ -178,9 +178,14 @@ class CreateAssociationViewmodel(
...
+              roles.values,
+              _uiState.value.members,
+              {
+                CurrentUser.associationUid = association.uid
+                userAPI.updateCurrentUserAssociationCache({ navActions.goFromCreateAsso() }, {})
+              },
+              {})
         },
         onFailure = { exception ->
           Log.e("CreateAssoViewModel", "Failed to add asso: ${exception.message}")
```
Comment (SidonieBouthors) [inline app/src/main/java/com/github/se/assocify/ui/screens/createAssociation/CreateAssociationViewmodel.kt:191]: Maybe adding at least some error logging for each onError would be better (or even error message / condition for what to do if relevant)

File: app/src/main/java/com/github/se/assocify/ui/screens/selectAssociation/SelectAssociationViewModel.kt [hunk 2]
```diff
@@ -49,26 +50,43 @@ class SelectAssociationViewModel(
...
+                              }
+                            },
+                            {})
+                      },
+                      {})
                 }
               },
               {})
         },
         {}) // TODO: handle joining assoc (non immediate approval)
```
Comment (SidonieBouthors) [inline app/src/main/java/com/github/se/assocify/ui/screens/selectAssociation/SelectAssociationViewModel.kt:89]: Make sure to remove debug error logs when the operation is a success :) And same as previously, adding error logging / messages for actual errors

**Review by SidonieBouthors — COMMENTED** [review]
Needs some more attention to dealing with errors, but the fix works great on my end, both for joining and creating a new assoc :)) Great work !

File: app/src/main/java/com/github/se/assocify/ui/screens/createAssociation/CreateAssociationViewmodel.kt (hunk 1 above)
Comment (SekoiaTree) [reply app/src/main/java/com/github/se/assocify/ui/screens/createAssociation/CreateAssociationViewmodel.kt:191]: There wasn't any previously, we're gonna add it in the next sprint

File: app/src/main/java/com/github/se/assocify/ui/screens/selectAssociation/SelectAssociationViewModel.kt (hunk 2 above)
Comment (SekoiaTree) [reply app/src/main/java/com/github/se/assocify/ui/screens/selectAssociation/SelectAssociationViewModel.kt]: Fixed!

**Review by SidonieBouthors — APPROVED** [review]
Looks good to me, works as expected, great bugfix :)

**Pull request merged**